community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/expressions/ToIntFunction.scala

```diff
@@ -37,16 +37,12 @@ case class ToIntFunction(a: Expression) extends NullInNullOutExpression(a) {
       v.longValue()
     case v: String =>
       try {
-        v.toLong
+        val d = BigDecimal(v)
```

Comment:
There's no real need to use BigDecimal. Just parse it as a Long, and catch the exception if it's wrong (that'll be rare enough that the exception wont be costly). Basically the same as is done here: https://github.com/neo4j/neo4j/blob/2.3/community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/ast/Literal.scala#L67-L81